playbooks/roles/rpc_pre_upgrade/tasks/main.yml

```diff
@@ -16,6 +16,36 @@
 # set backup_dir as a fact to ensure dir name doesn't change during run
 - set_fact:
     backup_dir: "{{ ansible_env.BKUPDIR |default(local_home+'/'+'rpc-upgrade-'+date_stamp) }}"
+    marker_dir: "{{ ansible_env.UPGRADE_LEAP_MARKER_FOLDER | default('/etc/openstack_deploy/upgrade-leap') }}"
```

Comment:
if we set marker dir here, other tasks cannot use the fact ('marker_dir') when they are executed individually. In this case, only main task has the scope of this static variable. I suggest moving this one to rpc_pre_upgrade/defaults/main.yml. And currently glance-cache-wipe task cannot be run separately, or there would be one undefined variable.